user_preferences.php

```diff
@@ -54,9 +54,12 @@
 if (!empty($action)) {
     if ($factorid != 0) {
         $instance = \tool_mfa\plugininfo\factor::get_instance_from_id($factorid);
-        $factor = \tool_mfa\plugininfo\factor::get_factor($instance->factor);
-        $string = $factor->get_display_name().' - '.$instance->label;
-        echo $OUTPUT->notification(get_string('factor'.$action, 'tool_mfa', $string), 'notifysuccess');
+        // Confirm factor is valid for the accessing user.
```

Comment:
I'm reading this file and can't understand why we need $userid. Should we just remove all references?

Reply:
Yeah this is a hang over from the very early implementation. Im not sure what the original intent was besides excluding guests, but I agree that it seems unnecessary. Ill kill it off